pinecone/index_connection.go

```diff
@@ -3,8 +3,9 @@ package pinecone
 import (
 	"context"
 	"crypto/tls"
-	"fmt"
```

Comment:
Are you sure this is meant to be deleted?

Reply:
It was in my original PR because I removed the `target := fmt.Sprintf("%s:443", in.host)` which was the only usage in here previously.

Once rebasing on your changes though we have those `fmt.Errorf` calls with the input struct checking.